amethyst_core/src/named.rs

```diff
@@ -0,0 +1,45 @@
+use specs::{Component, DenseVecStorage, EntityBuilder, WriteStorage};
+use specs::world::LazyBuilder;
+
+/// A component that gives a name to an `Entity`
+pub struct Named {
+    pub name: &'static str,
+}
+
+impl Named {
+    pub fn new(name: &'static str) -> Self {
+        Named { name }
+    }
+}
+
+impl Component for Named {
+    type Storage = DenseVecStorage<Self>;
+}
+
+/// An easy way to name an `Entity` and give it a `Named` `Component`.
+pub trait WithNamed
+where
+    Self: Sized,
+{
+    /// Adds a name to the entity being built.
+    fn named(self, name: &'static str) -> Self;
+}
+
+impl<'a> WithNamed for EntityBuilder<'a> {
+    fn named(self, name: &'static str) -> Self {
+        // Unwrap: The only way this can fail is if the entity is invalid and this is used while creating the entity.
+        self.world
+            .system_data::<(WriteStorage<'a, Named>,)>()
+            .0
+            .insert(self.entity, Named::new(name))
+            .unwrap();
```

Comment:
Would expect with a message be better here than just an unwrap, (granted its been a while since Ive actually written a lot of rust so I could be wrong).

Reply:
This is what the rest of the builders do.  The only reason this would fail is because the entity was dead, and there's no way it's dead while it's still being initialized.

That being said, some kind of comment explaining that could go well here.

Reply:
hmmmmmmmmmmm. This was already there?
```
// Unwrap: The only way this can fail is if the entity is invalid and this is used while creating the entity.
```

Reply:
haha thanks I wasn't paying very close attention

Reply:
Ok, yeah was just making sure, I've always been very cautious around leaving unwraps in released code but I wasn't sure if this was right or not.

Reply:
Talking about unwraps, anyone can check on the issue related to error crates. I would like to have a better system to handle errors than making a ton of enums for them, and nobody seems to know which crate to choose yet.

Reply:
@jojolepro open an issue unless there is already one, so that way we can discuss there.

Reply:
There is one already https://github.com/amethyst/amethyst/issues/460